src/main/java/org/openstreetmap/atlas/checks/flag/FlaggedPolyline.java

```diff
@@ -24,40 +27,35 @@
 {
     private static final long serialVersionUID = -1184306312148054279L;
 
+    private final AtlasItem atlasItem;
     private final String country;
     private final PolyLine polyLine;
     private final Map<String, String> properties;
 
     /**
      * Default constructor
      * 
-     * @param object
-     *            the {@link AtlasObject} to flag
+     * @param atlasItem
+     *            the {@link AtlasItem} to flag
      */
-    public FlaggedPolyline(final AtlasObject object)
+    public FlaggedPolyline(final AtlasItem atlasItem)
     {
-        if (object instanceof AtlasItem)
+        this.atlasItem = atlasItem;
+        if (atlasItem instanceof Area)
         {
-            if (object instanceof Area)
-            {
-                // An Area's geometry doesn't include the end (same as start) location. To close the
-                // area boundary, we need to add the end location manually.
-                final List<Location> geometry = Iterables
-                        .asList(((AtlasItem) object).getRawGeometry());
-                geometry.add(geometry.get(0));
-                this.polyLine = new PolyLine(geometry);
-            }
-            else
-            {
-                this.polyLine = new PolyLine(((AtlasItem) object).getRawGeometry());
-            }
+            // An Area's geometry doesn't include the end (same as start) location. To close the
+            // area boundary, we need to add the end location manually.
+            final List<Location> geometry = Iterables
+                    .asList(((AtlasItem) atlasItem).getRawGeometry());
+            geometry.add(geometry.get(0));
+            this.polyLine = new PolyLine(geometry);
```

Comment:
```this.polyLine= ((Area) atlasItem).asPolygon().closedLoop()```

Reply:
Good call, this wasn't my code, just taking it out of an if statement. That should be cleaner.

Reply:
Oh, I see how I can do it. I do think we should have polygons be polygons, but I can satisfy your suggestion by having `.closedLoop()` feed into constructing a `PolyLine`.

Reply:
Ok, done. I added a note that it's unclear why we only have FlaggedPolyline, and no FlaggedPolygon, since this is strange...